src/hooks/useTransactions.tsx

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import { ethers } from "ethers";
+
+type Transaction = ethers.providers.TransactionResponse;
+type TransactionReceipt = ethers.providers.TransactionRequest;
+type TransactionStatus = "idle" | "pending" | "resolved" | "rejected";
+
+type TransactionState = {
+  transaction?: Transaction;
+  status?: TransactionStatus;
+  error?: Error;
+};
+
+enum ActionType {
+  ERROR,
+  UPDATE,
+}
+
+type Action =
+  | {
+      type: ActionType.UPDATE;
+      payload: TransactionState;
+    }
+  | {
+      type: ActionType.ERROR;
+      payload: Pick<TransactionState, "error">;
+    };
+function reducer(state: TransactionState, action: Action) {
```

Comment:
theres no way as i see it to clear a transaction out once its errored or resolved. not sure if this is the intent to have it only ever manage a single transaction, or make subsequent transactions overwrite the previous inflight tx